indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -693,12 +707,17 @@ public long push() throws IOException
               try (ZipInputStream in = new ZipInputStream(fileSystem.open(zip, 1 << 13))) {
                 for (ZipEntry entry = in.getNextEntry(); entry != null; entry = in.getNextEntry()) {
                   final String fileName = entry.getName();
+                  final String outputPath = outDir.getAbsolutePath() + File.separator + fileName;
+
+                  CompressionUtils.validateZipOutputFile(
+                      zip.getName(),
+                      new File(outputPath),
+                      outDir
+                  );
+
                   try (final OutputStream out = new BufferedOutputStream(
-                      new FileOutputStream(
-                          outDir.getAbsolutePath()
-                          + File.separator
-                          + fileName
-                      ), 1 << 13
+                      new FileOutputStream(outputPath),
+                      1 << 13
```

Comment:
The same size is used in BufferedOutputStream by default in the constructor with lesser params, so suggested to not pass this

Reply:
I changed this to use the other constructor